include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +512,74 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class RandomAccessIterator>
+auto mode(RandomAccessIterator first, RandomAccessIterator last) -> std::vector<typename std::iterator_traits<RandomAccessIterator>::value_type>
+{
+    using Z = typename std::iterator_traits<RandomAccessIterator>::value_type;
+    using Size = typename std::iterator_traits<RandomAccessIterator>::difference_type;
+
+    std::vector<Z> sorted(first, last);
```

Comment:
There is an idiom in the standard library that sorting operations do not copy their input, they mutate it. See `std::nth_element`.

There might be a case to add a `sorted_mode` call which takes sorted data by `const &`, and this can forward to that call. I've found in statistical workflows, there are lots of things that require sorted data, so it's not a bad assumption that sorting is part of a larger workflow.